### PR TITLE
(fix): add `__version__`

### PR DESCRIPTION
**Pull request by LDeakin**

### CODE REVIEW

**Review by ilan-gold — APPROVED** [review]
Ok I should have just tried this.  It does work, and must be coming from the build process but don't really understand how.

**Comment by flying-sheep** [comment]
`importlib.metadata.version()` gets the version from an *installed* package’s metadata. It uses the import machinery to find the installed package’s “.dist-info/METADATA” file that contains all metadata like name, version, license, dependencies, …
So once you `hatch run ...` or `maturin develop` or `pip install -e .`, you get a special “editable” version of your package+metadata installed into a virtualenv. And calling `importlib.metadata.version()` inside of the package, the code is executed from that venv where it’s installed, and doing so finds that metadata and returns its value.

To explain everything else:

To build a Python package, one needs a build backend.
We currently use maturin directly as a build backend.
Having VCS versioning requires the build backend to support that.
Maturin doesn’t, and currently you can’t use maturin as a plugin for another build backend (https://github.com/PyO3/maturin/issues/1419), so you can’t leverage their capabilities (e.g. VCS versioning)

So we currently can’t do that. But:

Maturin derives the Python package’s version number from `Cargo.toml`. And for Rust packages exists *very* nice relase automation in the form of [release-plz](https://release-plz.ieni.dev/).

**Comment by LDeakin** [comment]
Since it all depends on the `Cargo.toml` package version anyway, I changed this to just use the `CARGO_PKG_VERSION`. [This is what `polars` does](https://github.com/pola-rs/polars/blob/50c1135c9b490528555142ef6fabf1b85b17b9ca/py-polars/src/lib.rs#L394-L395).

File: python/zarrs/__init__.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 from .pipeline import ZarrsCodecPipeline as _ZarrsCodecPipeline
 from .utils import CollapsedDimensionError, DiscontiguousArrayError
-
+from ._internal import __version__ as __version__
```
Comment (flying-sheep) [inline python/zarrs/__init__.py:5]:
```suggestion
from ._internal import __version__
```

**Comment by flying-sheep** [comment]
That’s 300μs faster I guess, and also not complicated, so either works I guess.

**Pull request merged**